manifests/init.pp

```diff
@@ -47,9 +78,48 @@
     }
   }
 
-  package { $package_name:
-    ensure   => $ensure,
-    provider => $provider,
+  case $manage_repo {
```

Comment:
I had added some in `spec/classes/curator_repo_spec.rb`. Did you have more in mind?